crates/client/devnet/src/lib.rs

```diff
@@ -388,8 +388,11 @@ mod tests {
 
         assert_eq!(res.class_hash, calculated_class_hash);
 
-        chain.block_production.set_current_pending_tick(1);
-        chain.block_production.on_pending_time_tick().unwrap();
+        let rt = tokio::runtime::Runtime::new().unwrap();
+        rt.block_on(async {
+            chain.block_production.set_current_pending_tick(1);
+            chain.block_production.on_pending_time_tick().await.unwrap();
+        });
```

Comment:
Why are you spawning a tokio runtime instead of using `#[tokio::test]`?

Reply:
this is because we are already using a fixture which creates it's own runtime and when we do tokio::test it's creating another runtime inside a runtime, which is a issue

Reply:
Which fixture is responsible for this? This still feels strange.

Reply:
my bad actually, so here is the thought process I followed in order to reach to the current scenerio:

1. code was changed in the on_pending_block_time in order to do dynamic block closing, so as soon as the limit is reached, we close the close, so in the code the close_block_time function was called which is async function.
2. now the pending_block_time has to async as well, so I updated the tests with tokio::test as well
3. when I did so, now within the test, we have this fixture which creates its own runtime, which is not allowed because tokio::test creates it's own runtime. 
4. so to solve this, I removed the tokio::test and created runtime for the specific part of the code